### PR TITLE
add fm.cancel.Ripcord

### PR DESCRIPTION
**Pull request by rayes0**
### Please confirm your submission meets all the criteria

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
  - network access, audio access, xdg-videos, xdg-pics, xdg-download for media upload as well as drag and drop
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://cancel.fm/ripcord/shareware-redistribution/
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: fm.cancel.Ripcord.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  - --filesystem=xdg-download
+
+command: Ripcord
+
+modules:
+  - name: Ripcord
+    buildsystem: simple
+    sources:
+      - type: file
+        url: https://cancel.fm/dl/Ripcord-0.4.29-x86_64.AppImage
```
Comment (hfiguiere) [inline fm.cancel.Ripcord.yaml:22]: you build an aarch64 flatpak with a x86_64 binary

https://github.com/flathub/flathub/wiki/App-Maintenance#only-build-on-x86_64

File: fm.cancel.Ripcord.yaml [hunk 2]
```diff
@@ -0,0 +1,38 @@
+app-id: fm.cancel.Ripcord
+runtime: org.freedesktop.Platform
+runtime-version: '22.08'
+sdk: org.freedesktop.Sdk
+finish-args:
+  - --socket=x11
+  - --socket=wayland
```
Comment (hfiguiere) [inline fm.cancel.Ripcord.yaml:7]: does it support wayland?

File: fm.cancel.Ripcord.yaml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+          url-pattern: "(https://cancel.fm/dl/Ripcord-([\\d\\.-]+)-x86_64.AppImage)"
+    build-commands:
+      - chmod +x ./Ripcord.AppImage
+      - ./Ripcord.AppImage --appimage-extract
+      - install -dm 755 /app/bin
+      - mv ./squashfs-root/Ripcord ./squashfs-root/lib ./squashfs-root/plugins ./squashfs-root/translations /app/bin
+      - mv ./squashfs-root/twemoji.ripdb ./squashfs-root/qt.conf /app/bin
+      - install -Dm 644 ./squashfs-root/Ripcord_Icon.png /app/share/icons/hicolor/scalable/apps/fm.cancel.Ripcord.png
+      - install -Dm 644 ./squashfs-root/Ripcord.desktop /app/share/applications/fm.cancel.Ripcord.desktop
+      - desktop-file-edit --set-key="Icon" --set-value="fm.cancel.Ripcord" /app/share/applications/fm.cancel.Ripcord.desktop
```
Comment (hfiguiere) [inline fm.cancel.Ripcord.yaml:38]: I'm gonna guess that there is no AppStream file. I't mandatory.

File: fm.cancel.Ripcord.yaml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+          url: https://cancel.fm/ripcord/index.html
+          version-pattern: "Ripcord Linux ([\\d\\.-]+) \\(AppImage\\)"
+          url-pattern: "(https://cancel.fm/dl/Ripcord-([\\d\\.-]+)-x86_64.AppImage)"
+    build-commands:
+      - chmod +x ./Ripcord.AppImage
+      - ./Ripcord.AppImage --appimage-extract
+      - install -dm 755 /app/bin
+      - mv ./squashfs-root/Ripcord ./squashfs-root/lib ./squashfs-root/plugins ./squashfs-root/translations /app/bin
+      - mv ./squashfs-root/twemoji.ripdb ./squashfs-root/qt.conf /app/bin
+      - install -Dm 644 ./squashfs-root/Ripcord_Icon.png /app/share/icons/hicolor/scalable/apps/fm.cancel.Ripcord.png
```
Comment (hfiguiere) [inline fm.cancel.Ripcord.yaml:36]: this is a lie. PNG is NOT scalable.

**Comment by hfiguiere** [comment]
bot, build fm.cancel.Ripcord

File: fm.cancel.Ripcord.metainfo.xml [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+<component type="desktop-application">
+  <id>fm.cancel.Ripcord</id>
+  <name>Ripcord</name>
+  <summary>Alternative desktop chat client for Slack (and Discord)</summary>
+
+  <url type="homepage">https://cancel.fm/ripcord//</url>
+  <url type="bugtracker">https://dev.cancel.fm/issues</url>
+
+  <metadata_license>CC0-1.0</metadata_license>
+  <project_license>GPL-3.0-or-later</project_license>
```
Comment (hfiguiere) [inline fm.cancel.Ripcord.metainfo.xml:11]: that's not what you linked too in your submission.

**Comment by hfiguiere** [comment]
bot, build fm.cancel.Ripcord

File: fm.cancel.Ripcord.yaml (hunk 2 above)
Comment (hfiguiere) [reply fm.cancel.Ripcord.yaml:7]: you didn't answer the question.

File: fm.cancel.Ripcord.yaml (hunk 3 above)
Comment (hfiguiere) [reply fm.cancel.Ripcord.yaml:38]: and it's not installed either.

File: fm.cancel.Ripcord.yaml (hunk 2 above)
Comment (rayes0) [reply fm.cancel.Ripcord.yaml:7]: oh yeah, not it doesn't, I forgot to add it to the previous commit

**Comment by hfiguiere** [comment]
bot, build fm.cancel.Ripcord

**Comment by barthalion** [comment]
bot, build fm.cancel.Ripcord

File: fm.cancel.Ripcord.metainfo.xml [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+  <url type="homepage">https://cancel.fm/ripcord//</url>
+  <url type="bugtracker">https://dev.cancel.fm/issues</url>
+
+  <metadata_license>CC0-1.0</metadata_license>
+  <project_license>LicenseRef-proprietary</project_license>
+  <content_rating type="oars-1.1"/>
+
+  <description>
+    <p>Ripcord is a desktop chat client for group-centric services like Slack and Discord. It provides a traditional compact desktop interface designed for power users. It's not built on top of web browser technology: it has a small resource footprint, responds quickly to input, and gets out of your way. Shareware is coming back, baby.</p>
+    <p>Ripcord is developed independently, and is <b>not affiliated with Slack or Discord.</b></p>
```
Comment (barthalion) [inline fm.cancel.Ripcord.metainfo.xml:16]:
```suggestion
    <p>Ripcord is developed independently, and is not affiliated with Slack or Discord.</p>
```

This is broken right now.

**Comment by hfiguiere** [comment]
bot, build fm.cancel.Ripcord

**Pull request closed without merging**